test/routes.js

```diff
@@ -106,6 +107,118 @@ test('GET route', async (t) => {
   })
 })
 
+test('server should return 400 on validation error', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      hello: String
+    }
+  `
+
+  app.register(GQL, {
+    schema,
+    resolvers: {
+      Query: {
+        hello: () => {}
+      }
+    }
+  })
+
+  const query = `
+    query {
+      hello
+      hello2
+      hello3
+    }
+  `
+
+  const response = await app.inject({
+    method: 'POST',
+    url: '/graphql',
+    body: { query }
+  })
+
+  t.equal(response.statusCode, 400)
+  t.matchSnapshot(JSON.stringify(JSON.parse(response.body), null, 2))
+})
+
+test('server should return 500 on missing operation name for multiple operations', async (t) => {
+  const app = Fastify()
+  const schema = `
+    type Query {
+      hello: String
+    }
+
+    type Mutation {
+      world: String
+    }
+  `
+
+  app.register(GQL, {
+    schema,
+    resolvers: {
+      Query: {
+        hello: () => {}
+      },
+      Mutation: {
+        world: () => {}
+      }
+    }
+  })
+
+  const query = `
+    query Hello {
+      hello
+    }
+
+    mutation World {
+      world
+    }
+  `
+
+  const response = await app.inject({
+    method: 'POST',
+    url: '/graphql',
+    body: { query }
+  })
+
+  t.equal(response.statusCode, 500)
+  t.matchSnapshot(JSON.stringify(JSON.parse(response.body), null, 2))
+})
+
+test('server should return 200 on graphql errors', async (t) => {
```

Comment:
According to the `express-graphql`, it depends on the field's type:

- If the field can be null -> 200: [source](https://github.com/graphql/express-graphql/blob/master/src/__tests__/http-test.js#L1131-L1159)
- If it's set to be NonNull, and the field throws -> 500 [source](https://github.com/graphql/express-graphql/blob/master/src/__tests__/http-test.js#L1161-L1196)

Reply:
I've found an issue with missing `data` field inside Error response and after fixing, it works as expected.

Same implementation inside `exporess-graphql`
https://github.com/graphql/express-graphql/blob/master/src/index.js#L366-L368